### PR TITLE
nixos/pantheon, nixos/cinnamon: Drop qgnomeplatform

### PR DESCRIPTION
**Pull request by bobby285271**
## Description of changes

https://github.com/NixOS/nixpkgs/pull/245166#issuecomment-1674855572

The upstream is officially [not maintained](https://github.com/FedoraQt/QGnomePlatform/), and qadwaitadecoration does not seem to be xorg-related. We should have proper shadows when using server side decoration.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/x11/desktop-managers/cinnamon.nix [hunk 1]
```diff
@@ -212,11 +212,6 @@ in
       programs.bash.vteIntegration = mkDefault true;
       programs.zsh.vteIntegration = mkDefault true;
 
-      # Harmonize Qt applications under Cinnamon
-      qt.enable = true;
-      qt.platformTheme = "gnome";
-      qt.style = "adwaita";
```
[the remark this answers is not in the document]
Comment (bobby285271) [reply nixos/modules/services/x11/desktop-managers/cinnamon.nix:218]: I think there is gtk3 the platformtheme in qtbase, I don't think this needs to be specified and you are all set if opening a Qt app uses your icon theme set in cinnamon-settings and opening a file from e.g. kate uses a native GTK file chooser.

But there is no gtk3 the style, I don't think Qt is willing to do it, the default style will be Fusion. Which means this cannot make your Qt apps look and feel matches your GTK theme.

The gtk2 platformtheme & style are [third party](https://github.com/trialuser02/qt6gtk2) projects, that can make Qt apps look and feel matches your GTK theme, but implementation will actually depend on gtk2.

**Comment by bobby285271** [comment]
@ofborg test pantheon cinnamon

File: nixos/modules/services/x11/desktop-managers/cinnamon.nix (hunk 1 above)
[the remark this answers is not in the document]
Comment (mkg20001) [reply nixos/modules/services/x11/desktop-managers/cinnamon.nix:218]: I don't feel qualified enough to give a last word, but the matching theme feature sounds nice.

[the remark this answers is not in the document]
Comment (bobby285271) [reply nixos/modules/services/x11/desktop-managers/cinnamon.nix:218]: > the matching theme feature sounds nice

Ugh, so to be clear shall we apply the gtk2 platformtheme & style by default or do you think asking users to do that in their NixOS config is okay? :joy: We should have fair reasons to do either

[the remark this answers is not in the document]
Comment (mkg20001) [reply nixos/modules/services/x11/desktop-managers/cinnamon.nix:218]: by default sounds fine, maybe disable if other  desktop environments are enabled

[the remark this answers is not in the document]
Comment (bobby285271) [reply nixos/modules/services/x11/desktop-managers/cinnamon.nix:218]: > maybe disable if other desktop environments are enabled

Not sure if I get your idea right, I think both gtk2 and fusion are used in at least one desktops right now I think (e.g. budgie for gtk2 and xfce for fusion).

[the remark this answers is not in the document]
Comment (bobby285271) [reply nixos/modules/services/x11/desktop-managers/cinnamon.nix:218]: > by default sounds fine

Okay I filed #249712

**Pull request merged**